### PR TITLE
fix: improve boolean logic to verify proposal type

### LINKED ISSUE #2111: [RFP] Missing RFP label on newly created RFP proposal
missingrfpfb

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #2111 

### UI Changes Screenshot

**Before:** see issue.
**After:**

fixrfp

### CODE REVIEW

File: src/components/Proposal/Proposal.jsx [hunk 1]
```diff
@@ -203,7 +205,7 @@ const Proposal = React.memo(function Proposal({
               subtitle={
                 <Subtitle>
                   <Author username={username} url={authorURL} />
-                  {isRfp && (
+                  {isRfp && linkby && (
```
Comment (amassarwi) [inline src/components/Proposal/Proposal.jsx:208]: Why is `linkby` needed again here ? we have already in `isRfp` definition ? 🤔

Comment (tiagoalvesdulce) [reply src/components/Proposal/Proposal.jsx:208]: We need to verify if `linkby` is truthy in order to pass it to `<Event />`. Here:

https://github.com/decred/politeiagui/blob/e95dac9a11c8dc7c05bf71c33fccd9b15b7d0176/src/components/Proposal/Proposal.jsx#L209

We can have a proposal without a `linkby` in the case described by the issue (newly created proposal). As soon as we fetch this proposal info, the linkby will be populated.

**Comment by tiagoalvesdulce** [comment]
@victorgcramos answered as a reply to @amassarwi's comment.

File: src/components/Proposal/Proposal.jsx (hunk 1 above)
Comment (tiagoalvesdulce) [reply src/components/Proposal/Proposal.jsx:208]: Regarding the title, I opened the PR from the CLI and didn't notice it was so damn long lol. Sorry about that.

**Comment by victorgcramos** [comment]
Fair enough.

File: src/components/Proposal/Proposal.jsx (hunk 1 above)
Comment (amassarwi) [reply src/components/Proposal/Proposal.jsx:208]: Makes sense, thanks for clarifying.

**Pull request merged**